src/components/Square.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+const Square = ({
+  i,
+  squares,
+  setSquares,
+  remainingAlphabet,
+  setRemainingAlphabet,
+  errorMessage,
+  setErrorMessage,
+  maxNumberConsonants,
+}) => {
+  const vowels = ['A', 'E', 'I', 'O', 'U'];
+  const totalNumberOfConsonants = 27;
+  
+  const editInput = e => {
+    const newSquares = JSON.parse(JSON.stringify(squares));
+    const workRemainingAlphabet = JSON.parse(JSON.stringify(remainingAlphabet));
+    let newLetter = e.target.value.replace(/[^a-z]/gi, '').toUpperCase();
+    
+    // Check to see if you have reached the extent of your letter useage
+    let workErrorMessage = '';
+    if (
+      maxNumberConsonants < totalNumberOfConsonants - remainingAlphabet.length + 1 &&
+      vowels.indexOf(newLetter) === -1
+    ) {
+      workRemainingAlphabet.push('');
+      workErrorMessage = 'You have reached the extent of your letter useage... please start over';
+      newLetter = '';
+    }
+    setErrorMessage(workErrorMessage);
+    // Ensure input is a letter and if it is save it
+    // Ensure letter is available. If not generate ab error message
+    if (
+      workRemainingAlphabet.indexOf(e.target.value.toUpperCase()) === -1 &&
+      e.target.value !== '' &&
+      workErrorMessage === '' &&
+      vowels.indexOf(newLetter) === -1
+    ) {
+      workRemainingAlphabet.push('');
+      workErrorMessage = 'Letter is not available';
+      newLetter = '';
+    }
+    setErrorMessage(workErrorMessage);
+    // Add letter to available list if removed
+    if (
+      newSquares[i].letter !== '' &&
+      e.target.value === '' &&
+      vowels.indexOf(newSquares[i].letter) === -1
+    ) {
+      workRemainingAlphabet.push(newSquares[i].letter);
+    }
+    // if letter entered was not '' and was not a vowel, remove it from alphabet list
+    if (newLetter !== '') {
+      if (vowels.indexOf(newLetter) === -1) {
+        workRemainingAlphabet.splice(workRemainingAlphabet.indexOf(newLetter), 1);
+      }
+    }
+    // save state
+    newSquares[i].letter = newLetter;
+    setSquares(newSquares);
```

Comment:
I think a lot of this logic could be put into your custom hook in a new method called like addLetter or something. Ideally your custom hook shouldn't return `setSquares` directly. Instead it returns expected methods that set the square for you

Reply:
Done. I think.